modules/core/shared/src/main/scala/lucuma/core/enum/MagnitudeBand.scala

```diff
@@ -19,32 +27,40 @@ sealed abstract class MagnitudeBand(
   val shortName: String,
   val longName: String,
   val center: Wavelength,
-  val width: Int,
+  val width: Quantity[PosInt, Nanometer],
   val magnitudeSystem: MagnitudeSystem
-) extends Product with Serializable
+) extends Product with Serializable {
+  require(center.toPicometers.value >= width.value / 2)
+
+  def end: Wavelength = Wavelength(center.toPicometers + (width.value.value / 2).withRefinedUnit[Positive, Nanometer])
```

Comment:
Not a big deal here, but I _think_ you can avoid stripping the units off width for the division and then adding them back on by `center.toPicometers + width / 2.withRefinedUnit[Positive, Unitless]`.

Reply:
yes you can see
https://github.com/erikerlandson/coulomb/issues/211